services/api/src/messaging/players.service.ts

```diff
@@ -84,21 +84,26 @@ const setDefaultStatusForPlayers = (ws: GameWebSocket): void => {
 export const clearBoard = (ws: GameWebSocket): void => {
   gameService.clearBoard(ws.roomId);
   setDefaultStatusForPlayers(ws);
+  sendMessage(ws, MessageType.SetMyTurn, null);
   publish(ws.roomId, { type: MessageType.ClearBoard });
   publishBoard(ws.roomId);
   publishAllPlayers(ws.roomId);
 };
 
-export const checkIfUserAlreadyExists = (ws: GameWebSocket): void => {
+export const checkIfUserAlreadyExists = (
+  ws: GameWebSocket,
+  playerId: string,
+): void => {
   const players = getPlayers(ws.roomId);
-  const myTurn = gameService.findMoveByPlayerId(ws.roomId, players.get(ws)?.id);
+  const myTurn = gameService.findMoveByPlayerId(ws.roomId, playerId);
 
   if (myTurn) {
     players.set(ws, {
       ...players.get(ws),
       status: PlayerStatus.FigurePlaced,
     });
-    publish(ws.roomId, { type: MessageType.SetMyTurn, payload: myTurn });
+
+    sendMessage(ws, MessageType.SetMyTurn, myTurn);
```

Comment:
This action does not fit in this method.
I would expect a boolean returned from this method, instead it does not return anything and has side effects.

Reply:
Why do we need to have info about previous game. I think we just reset everything going into the next game.